src/modules/job-list/job_data.c

```diff
@@ -244,6 +244,11 @@ static int parse_jobspec_ntasks (struct job *job,
             && res[2].type != NULL && !strcmp (res[2].type, "core")
             && res[2].with == NULL)
             job->ntasks = res[0].count * count;
+        else if (streq (type, "core")
+                 && res[0].type != NULL && !strcmp (res[0].type, "slot")
+                 && res[1].type != NULL && !strcmp (res[1].type, "core")
+                 && res[1].with == NULL)
+            job->ntasks = res[0].count * count;
```

Comment:
I'm not sure this is correct. With `per-resource.type=core` the number of tasks is the number of total cores (not slots) multiplied by `per-resource.count`. E.g.
```
$ flux mini run -c 2 -o per-resource.type=core -o per-resource.count=2 hostname
asp
asp
asp
asp
ƒ(s=1,d=0,builddir) grondo@asp:~/git/flux-core.git$ flux jobs -a
       JOBID USER     NAME       ST NTASKS NNODES  RUNTIME NODELIST
    ƒdgFNVaF grondo   hostname   CD      2      1   0.048s asp
```

`NTASKS` here should be 4 not 2.

Reply:
There's another case that doesn't work, but this one is trickier. Sometimes we don't know the number of tasks until the job has resources assigned, e.g.
```
$ flux mini run -N1 --tasks-per-core=2 hostname
asp
asp
asp
asp
asp
asp
asp
asp
ƒ(s=1,d=0,builddir) grondo@asp:~/git/flux-core.git$ flux jobs -ac1
       JOBID USER     NAME       ST NTASKS NNODES  RUNTIME NODELIST
   ƒ5BuNW95Z grondo   hostname   CD      1      1   0.055s asp
```

`NTASKS` should be 8 here because 1 node with 4 cores was allocated exclusively with `--tasks-per-core=2`. In this case job-list doesn't know how many total tasks there will be until the job is allocated resources, so IMO NTASKS should remain unset until _R_ is available.

Reply:
thanks, i clearly misunderstood the parsing of the jobspec + per-resource=core.  Lemme work on these.